src/neo4j_genai/retrievers/text_to_cypher.py

```diff
@@ -0,0 +1,104 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import logging
+from typing import Optional
+
+import neo4j
+from pydantic import ValidationError
+
+from neo4j_genai.llm import LLM
+from neo4j_genai.prompts import TEXT_TO_CYPHER_PROMPT
+from neo4j_genai.retrievers.base import Retriever
+from neo4j_genai.schema import get_schema
+from neo4j_genai.types import (
+    LLMModel,
+    Neo4jDriverModel,
+    Neo4jSchemaModel,
+    TextToCypherRetrieverModel,
+    TextToCypherSearchModel,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class TextToCypherRetriever(Retriever):
```

Comment:
Could you rename this to `Text2CypherRetriever`, it's a convention that's already established for Cypher but also in `Text2SQL` functions.

Reply:
Done!